### PR TITLE
refine_refactor_invoice

### PR DESCRIPTION
**Pull request by ChrOertlin**
### Added

- Tests for invoice model
- Minor refactoring of `get_invoice_by_id` to `get_invoice_by_entry_id`

### How to prepare for test

- [x] Ssh to relevant server (depending on type of change)
- [x] Use statusdb-stage in cigrid
- [x] Paxa the environment: `paxa`
    ```

### How to test

- [ ] Do make sure tests run
- [ ] Do make sure that invoice portal behaves as expected see below

### Expected test outcome


- [ ] Take a screenshot and attach or copy/paste the output. See below

## Review

- [ ] Tests executed by
- [ ] "Merge and deploy" approved by
Thanks for filling in who performed the code review and the test!

### This [version](https://semver.org/) is a

- [ ] **MAJOR** - when you make incompatible API changes
- [ ] **MINOR** - when you add functionality in a backwards compatible manner
- [x] **PATCH** - when you make backwards compatible bug fixes or documentation/instructions

## Implementation Plan

- [ ] Document in ...
- [ ] Deploy this branch on ...
- [ ] Inform to ...

### CODE REVIEW

File: tests/store/api/test_find_business_data.py [hunk 1]
```diff
@@ -597,3 +597,29 @@ def test_is_case_external_false(base_store: Store, case_obj: Family, sample_id:
...
+    """Test that invoices can be fetched by invoice id."""
+    # GIVEN a database with two invoices of which one has attributes
+
+    # WHEN fetching the invoice by invoice id
+    invoice: Invoice = store_with_an_invoice_with_and_without_attributes.get_invoice_by_entry_id(
+        entry_id=1
+    )
+
+    # THEN one invoice should be returned
+    assert invoice and invoice.id == 1
```
Comment (diitaz93) [inline tests/store/api/test_find_business_data.py:625]:
```suggestion
    assert isinstance(invoice,Invoice)
    assert invoice.id == 1
```

**Review by diitaz93 — CHANGES_REQUESTED** [review]
👍

**Review by diitaz93 — APPROVED** [review]
👍

**Comment by ChrOertlin** [comment]
1. Can retrieve invoice using entry id:
image

2. Can create new invoice entry 2023-03-17
image

**Pull request merged**